app/bridge/smuggler.js

```diff
@@ -215,47 +310,90 @@
         // But for now we are lazy and start the server as soon the client
         // is connected.
 
-        if (!hasServerForContract(contract)) {
-          var config = getConfigForContract(contract)
-          var server = new kConfigTypes[config.type](config.url);
+        if (hasServerReadyForContract(contract)) {
+          var server = getServerForContract(contract);
+          registerClientToServer(contract, server, registration.uuid);
+        } else {
+          startServer(contract);
+        }
```

Comment:
Because I had race condition where the client tried to register to the server before the code in `server.js` had run, thus sending a register request before the server would be able to answer them.